nxc/modules/snipped.py

```diff
@@ -0,0 +1,126 @@
+import ntpath
+import os
+from impacket import smb, smb3
+from os.path import join, exists
+from dploot.lib.smb import DPLootSMBConnection
+from dploot.lib.target import Target  
+from nxc.paths import NXC_PATH
+
+class NXCModule:
+
+    name = "snipped"
+    description = "Downloads screenshots taken by the (new) Snipping Tool."
+    supported_protocols = ["smb"]
+    opsec_safe = True
+    multiple_hosts = True
+
+    def __init__(self):
+        self.context = None
+        self.module_options = None
+
+    def options(self, context, module_options):
+        """
+        USERS           Download only specified user(s); format: -o USERS=user1,user2,user3
+        """
+        self.context = context
+        self.screenshot_path_stub = r"Pictures\Screenshots"
+        self.users = module_options["USERS"].split(",") if "USERS" in module_options else None
+
+    def on_admin_login(self, context, connection):
+        self.context = context
+        self.connection = connection
+        self.share = "C$"
+
+        self.hostname = connection.hostname
+        self.domain = connection.domain
+        self.username = connection.username
+        self.password = getattr(connection, "password", "")
+        self.host = connection.host
+        self.kerberos = connection.kerberos
+        self.lmhash = getattr(connection, "lmhash", "")
+        self.nthash = getattr(connection, "nthash", "")
+        self.aesKey = connection.aesKey
+        self.use_kcache = getattr(connection, "use_kcache", False)
+
+        target = Target.create(
+            domain=self.domain,
+            username=self.username,
+            password=self.password,
+            target=self.hostname + "." + self.domain if self.kerberos else self.host,
+            lmhash=self.lmhash,
+            nthash=self.nthash,
+            do_kerberos=self.kerberos,
+            aesKey=self.aesKey,
+            no_pass=True,
+            use_kcache=self.use_kcache,
+        )
+
+        dploot_conn = self.upgrade_connection(target=target, connection=connection.conn)
```

Comment:
Is there a reason for a DPLoot connection? Looks like it is only used to listPaths, which can be also done with the `connection.conn` object.

Reply:
Hey thanks for the rework and sorry for my delayed responses. 

From my testing the Path `%HOMEPATH%/Pictures` is always the screenshot folder, despite the OS being set to a different language (probably hardcoded by MS to ensure stability when looking for stuff like that). Therefore, i would prefer not iterating recursively. That can be really time consuming and the spider_plus module already does exactly that. 

The rest is working, so when that is resolved we should be good to go

Reply:
Missunderstanding from my side. The code looks like it just enumerates the users with a screenshot folder, not recursively any folder (and depth) for potential screenshots. So that looks good to me 👍